nemo/collections/common/parts/preprocessing/manifest.py

```diff
@@ -61,17 +67,41 @@ def item_iter(
     if isinstance(manifests_files, str):
         manifests_files = [manifests_files]
 
+    num_manifests = len(manifests_files)
+    if isinstance(data_prefix, str):
+        # use the same prefix for all manifests
+        data_prefix = [data_prefix] if num_manifests == 1 else [data_prefix]*num_manifests
+    elif data_prefix is None:
+        # append no prefix
+        data_prefix = [""] * num_manifests
+    
+    num_prefix = len(data_prefix)
+    if num_prefix > 1 and num_prefix != num_manifests:
+        raise ValueError(
+            f"The number of manifests and prefixes must match, unless a single or none prefix is provided." \
+             f"Got {num_manifests} manifests with {num_prefix} prefix(es)."
+        )
+
     if parse_func is None:
         parse_func = __parse_item
 
     k = -1
-    for manifest_file in manifests_files:
+    for manifest_file, prefix in zip(manifests_files, data_prefix):
+        if not Path(manifest_file).is_file():
```

Comment:
This step is slow - it will have an effect when passing ASRSet to do this repeatedly